genienlp/data_utils/numericalizer/simple.py

```diff
@@ -120,33 +128,22 @@ def encode_single(self, minibatch, decoder_vocab, device=None, max_length=-1):
             max_len = max(len(x[0]) for x in minibatch)
         else:
             max_len = self.fix_length
-        padded = []
+        examples = []
         lengths = []
         numerical = []
         decoder_numerical = []
         for tokens, _mask in minibatch:
-            if self.pad_first:
-                padded_example = [self.pad_token] * max(0, max_len - len(tokens)) + \
-                                 [self.init_token] + \
-                                 list(tokens[:max_len]) + \
-                                 [self.eos_token]
-            else:
-                padded_example = [self.init_token] + \
-                                 list(tokens[:max_len]) + \
-                                 [self.eos_token] + \
-                                 [self.pad_token] * max(0, max_len - len(tokens))
+            example = [self.init_token] + \
+                                list(tokens[:max_len]) + \
+                                [self.eos_token]
 
-            padded.append(padded_example)
-            lengths.append(len(padded_example) - max(0, max_len - len(tokens)))
-
-            numerical.append([self.vocab.stoi[word] for word in padded_example])
-            decoder_numerical.append([decoder_vocab.encode(word) for word in padded_example])
+            examples.append(example)
+            lengths.append(len(example))
 
-        length = torch.tensor(lengths, dtype=torch.int32, device=device)
-        numerical = torch.tensor(numerical, dtype=torch.int64, device=device)
-        decoder_numerical = torch.tensor(decoder_numerical, dtype=torch.int64, device=device)
+            numerical.append([self.vocab.stoi[word] for word in example])
+            decoder_numerical.append([decoder_vocab.encode(word) for word in example])
 
-        return SequentialField(length=length, value=numerical, limited=decoder_numerical)
+        return SequentialField(length=lengths, value=numerical, limited=decoder_numerical)
```

Comment:
Unrelated to this PR, but I'm getting the feeling that, when we kill GloVe (which we should do soon), we should also remove the whole Numericalizer abstraction and go straight to Huggingface's interface for padding. It will be more correct with models that do all sorts of weird padding things I bet.

Reply:
Yeah, we will get to that right after the `transformers` version bump, which will clean up their tokenization and padding interfaces.